packages/central-server/app/admin/patientMerge/mergePatient.js

```diff
@@ -381,6 +400,9 @@ export async function mergePatient(models, keepPatientId, unwantedPatientId) {
       visibilityStatus: VISIBILITY_STATUSES.MERGED,
     });
 
+    // See the function's documentation for more details on why this is needed
+    await updateDependentRecordsForResync(models, unwantedPatientId);
```

Comment:
We also need to do this in the `PatientMergeMaintainer` scheduled task.

Quite wild how much repeated code/responsibility there is across the two...but we should handle that another day.